src/client/clouds.cpp

```diff
@@ -28,11 +28,10 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 #include "settings.h"
 #include <cmath>
 
-
-// Menu clouds are created later
 class Clouds;
-Clouds *g_menuclouds = NULL;
+IWritableShaderSource *g_menu_shader_source = NULL;
```

Comment:
this can be moved to clientlauncher.cpp and doesn't need to be public
also `unique_ptr`

Reply:
I tried unique_ptr, but I still had to delete the thing manually to get correct deletion order and it didn't work with forward-declaring IWritableShaderSource in clientlauncher.h

Reply:
but you don't need to declare it in clientlauncher.h?

Reply:
Ah, I see